internal/http/apitester.go

```diff
@@ -89,9 +88,13 @@ func (c *APITester) GetProject(ctx context.Context, hookID string) (apitest.Proj
 	return project, nil
 }
 
-func (c *APITester) GetEventResult(ctx context.Context, hookID string, eventID string) (apitest.TestResult, error) {
+func (c *APITester) GetEventResult(ctx context.Context, rateLimiter *rate.Limiter, hookID string, eventID string) (apitest.TestResult, error) {
+	if err := rateLimiter.Wait(context.Background()); err != nil {
```

Comment:
You should use the given context:
```suggestion
	if err := rateLimiter.Wait(ctx); err != nil {
```